src/query.c

```diff
@@ -292,6 +296,7 @@ QueryNode *NewVectorNode_WithParams(struct QueryParseCtx *q, VectorQueryType typ
   VectorQuery *vq = rm_calloc(1, sizeof(*vq));
   ret->vn.vq = vq;
   vq->type = type;
+  ret->opts.flags |= QueryNode_YieldsDistance;
```

Comment:
Why do we need this flag?

Reply:
This is part of the generic yield distance mechanism, the purpose is that every query node which is able to yield a distance field, will set this flag to true. We can discuss this approach further if you want.